cmd/object-api-utils.go

```diff
@@ -338,7 +442,7 @@ type ObjReaderFn func(inputReader io.Reader, h http.Header, cleanupFns ...func()
 // are called on Close() in reverse order as passed here. NOTE: It is
 // assumed that clean up functions do not panic (otherwise, they may
 // not all run!).
-func NewGetObjectReader(rs *HTTPRangeSpec, oi ObjectInfo, cleanUpFns ...func()) (
+func NewGetObjectReader(rs *HTTPRangeSpec, oi ObjectInfo, decompress bool, cleanUpFns ...func()) (
```

Comment:
I think this won't work we have to decompress all the time since the actual size per part will vary once you do GetObject() --> PutObject() 

GetObject() might be happening on an object which has variadic parts whereas PutObject() will create parts in 64MiB   order, it won't be optimal and might be confusing. 

It is better to decompress always just like encryption.

Reply:
Agreed

Reply:
Done